### DISCUSSION TITLE
[General] Is there any example/guide/tutorial on how to use Python SDK to interact with specific smart contract?

### DISCUSSION THREAD

**Discussion by cryptoleadxyz** [zkSync-Community-Hub/zksync-developers#393] [SDKs] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

I have read through the zkSync documentation which has example on Python SDK for account automation such as transferring funds between accounts. However, are there an guide or examples that show how I can use Python SDK to interact with smart contract such as swapping or providing LP to a DEX?

### What is the specific issue or error you're encountering?

Not an issue per se but request for learning resources.

### Can you share the error messages or logs you're receiving, if any?

n/a

### Have you made any recent changes to the contract before encountering this issue?

n/a

### Are there any external libraries or contracts that your contract interacts with?

n/a

### Can you provide the relevant portions of your contract code where the issue is occurring?

n/a

### Have you tried to isolate the problem, and if so, what were the results?

n/a

### What steps have you already taken to try to resolve the issue?

n/a

**Comment by dgeyston** [comment]
Check the zkSync Python SDK documentation for examples and guides on interacting with smart contracts, such as swapping tokens or providing liquidity to a DEX.

**Comment by mariyokaveen** [comment]
You can interact with smart contracts on zkSync using the Python SDK by deploying contracts, calling their functions, and querying their states. You may refer to the zkSync Python SDK documentation and explore examples provided there for guidance.

**Comment by dutterbutter** [comment, marked as answer]
@cryptoleadxyz there are some python examples here:
- https://github.com/zksync-sdk/zksync2-examples/tree/main/python

**Comment by bryanbrega** [comment]
The user is looking for learning resources on how to use the Python SDK to interact with smart contracts such as swapping or providing liquidity to a DEX on zkSync. They have read through the zkSync documentation and have found examples on account automation, but not on interacting with smart contracts. No error messages or logs are provided, and there have been no recent changes to the contract. There are no external libraries or contracts involved, and the relevant portions of the contract code are not provided. The user has not attempted to isolate the problem or resolve the issue yet.
